lib/modules/datasource/rpm/index.ts

```diff
@@ -118,7 +115,7 @@ export class RpmDatasource extends Datasource {
     packageName: string,
   ): Promise<ReleaseResult | null> {
     let response: HttpResponse<Buffer>;
-    let decompressedBuffer: Buffer;
+    let decompressStream: Readable;
     try {
       // primaryGzipUrl is a .gz file, need to extract it before parsing
       response = await this.http.getBuffer(primaryGzipUrl);
```

Comment:
we should use `got.stream` here, then you can pipe it directly without a memory buffer

Reply:
Likely yes, but:

1. There is a length check a few lines below: https://github.com/renovatebot/renovate/blob/096e7bcfd31bc9299269ce6ecb0c8df34ef5cbb1/lib/modules/datasource/rpm/index.ts#L126-L127 - how should it be implemented then? Checking `content-length` isn't exactly the same thing, as far as I understand.

2. It'll be harder to check the magic number in https://github.com/renovatebot/renovate/pull/37997

Reply:
Done